### DISCUSSION TITLE
Can't create MySQL procedures in release 24.0.3

### DISCUSSION THREAD

**Discussion by frelling** [dbeaver/dbeaver#26370] [General]
I am developing a new MySQL procedure.  The 'create procedure' statement for the current version of the sproc worked with no problems Friday.  Today the DBeaver update to version 24.0.3 was available.  After I installed the update the 'create procedure' statement fails. I had made some updates to the code but I reverted to Friday's version after I started getting errors from DBeaver.

At first it was complaining about a DELIMITER statement.  I removed that and now it is complaining about an EXIT HANDLER for SQLEXCEPTION statement, specifically the v_error_message = MESSAGE_TEXT... line:

DECLARE EXIT HANDLER FOR SQLEXCEPTION
      BEGIN
         GET DIAGNOSTICS CONDITION 1
         v_error_message = MESSAGE_TEXT, v_sqlstate = RETURNED_SQLSTATE, v_error_schema = SCHEMA_NAME, v_error_TABLE = TABLE_NAME, v_err_nbr = MYSQL_ERRNO;
         SELECT v_error_message, v_sqlstate, v_error_schema, v_error_table, v_err_nbr;
         SET v_updt_row_count = 1;
      END;

I can run the create procedure statement in a mysql client on a Linux server with no issues.

Anyone else running into this or have any insights?

Thanks in advance!

**Comment by E1izabeth** [comment]
You could try switching `Blank line is statement delimiter` setting to `Always`
![image](https://github.com/dbeaver/dbeaver/assets/28875055/be8457a0-d62c-4acb-ac2f-67ce518cf3f5)

**Comment by Asabeaaa** [comment]
Hello @frelling did you find a fix for this issue? I am using DBeaver version 24.3.3 and when i try to create a stored procedure and I put a semi colon after END it throws an error. But it works when I run it in another database client (Data Grip)

**Reply by frelling** [reply]
Hi @Asabeaaa,
My issue just went away after a couple of days, same as it showed up.  The only difference was that there wasn't a new patch release of DBeaver.  I never did find out what specifically caused the issued, nor what fixed it.  I tried the suggestion by E1izabeth and several things I found using Google, but none made a difference.

Did your problem just start happening, or is this the first time you are using DBeaver to create a sproc?  You are probably already doing something similar but when I create a sproc in DBeaver I temporarily set the delimiter to something other than ';' , which is the default delimiter for DBeaver.  For example:
USE some_schema;
DELIMITER $$
CREATE PROCEDURE some_schema.sproc_name
...
...
END$$
DELIMITER ;

And then I run all of the statements together.

Sorry I couldn't be of more help.

**Reply by E1izabeth** [reply]
@frelling 
If the proposed solution doesn't help, you can create an issue.
Also, as a workaround, such a query will be successfully executed if you select the whole query text and press Execute SQL Query (Ctrl+Enter). In this case, DBeaver doesn't try to be smarter than you and sends the whole selected text to the database as is.
